CASMINST-5068 - Script for section 3.1, 3.2 and 3.3 (#2182)
Cray-HPE/docs-csm: troubleshooting/known_issues/sms_services_health_checks.md

##  Software Management Services health checks

1. [SMS test execution](#31-sms-test-execution)
1. [Interpreting `cmsdev` Results](#32-interpreting-cmsdev-results)
1. [Known issues with SMS tests](#33-known-issues-with-sms-tests)

### 3.1 SMS test execution

The test in this section requires that the [Cray CLI is configured](#0-cray-command-line-interface) on nodes where the test is executed.

The following test can be run on any Kubernetes node (any master or worker node, but **not** the PIT node).

```bash
/usr/local/bin/cmsdev test -q all
```

- The `cmsdev` tool logs to `/opt/cray/tests/cmsdev.log`
- The -q (quiet) and -v (verbose) flags can be used to decrease or increase the amount of information sent to the screen.
  - The same amount of data is written to the log file in either case.

### 3.2 Interpreting `cmsdev` results

- If all checks passed, then the following will be true:

>   - The return code will be zero.

Reviewer: ```suggestion
  - The return code is zero.
```

>   - The final line of output will begin with `SUCCESS`.

Reviewer: ```suggestion
  - The final line of output begins with `SUCCESS`.
```

    - For example: `SUCCESS: All 7 service tests passed: bos, cfs, conman, crus, ims, tftp, vcs`

> - If one or more checks failed, then the following will be true:

Reviewer: ```suggestion
- If one or more checks failed, the following are true:
```

>   - The return code will be non-zero.

Reviewer: ```suggestion
  - The return code is non-zero.
```

>   - The final line of output will begin with `FAILURE` and will list which checks failed.

Reviewer: ```suggestion
  - The final line of output begins with `FAILURE` and lists the failed checks.
```

    - For example: `FAILURE: 2 service tests FAILED (conman, ims), 5 passed (bos, cfs, crus, tftp, vcs)`

>   - After remediating a test failure for a particular service, just that single service test can be re-run by replacing

Reviewer: ```suggestion
  - After remediating a test failure for a particular service, just that single service test can be rerun by replacing
```
